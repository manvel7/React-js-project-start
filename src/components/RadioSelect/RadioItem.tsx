import React, {FunctionComponent, InputHTMLAttributes, ReactNode} from "react"
import HiddenInput from "./HiddenInput/HiddenInput"
import RadioItemWrapper from "./ItemWrapper"

interface IProps extends InputHTMLAttributes<HTMLInputElement> {
  label?: ReactNode
  checked?: boolean
  error?: boolean
  disabled?: boolean
}

const RadioItem: FunctionComponent<IProps> = (props) => {
  const {label, checked, disabled, error} = props

  return (
    <RadioItemWrapper
      label={label}
      checked={checked}
      error={error}
      disabled={disabled}>
      <HiddenInput {...props} disabled={disabled} />
      <div className="radio-select__item" />
      {label && <span>{label}</span>}
    </RadioItemWrapper>
  )
}

export default RadioItem
