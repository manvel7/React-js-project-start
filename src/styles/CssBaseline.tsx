import {
  createGlobalStyle,
  css,
  DefaultTheme,
  GlobalStyleComponent,
  StyledProps
} from "styled-components"
import {mediaBreakpointDown} from "../utils/breakpoint"
import {getValue, toPercent} from "../utils/unit"
import {
  bodyMixin,
  smallMixin,
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin
} from "./typography"

interface IProps {}

const genericResetMixin = css<StyledProps<IProps>>`
  html,
  body,
  div,
  span,
  applet,
  object,
  iframe,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  blockquote,
  pre,
  a,
  abbr,
  acronym,
  address,
  big,
  cite,
  code,
  del,
  dfn,
  em,
  img,
  ins,
  kbd,
  q,
  s,
  samp,
  small,
  strike,
  strong,
  sub,
  sup,
  tt,
  var,
  b,
  u,
  i,
  center,
  dl,
  dt,
  dd,
  ol,
  ul,
  li,
  fieldset,
  form,
  label,
  legend,
  table,
  caption,
  tbody,
  tfoot,
  thead,
  tr,
  th,
  td,
  article,
  aside,
  canvas,
  details,
  embed,
  figure,
  figcaption,
  footer,
  header,
  hgroup,
  menu,
  nav,
  output,
  ruby,
  section,
  summary,
  time,
  mark,
  audio,
  video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  menu,
  nav,
  section {
    display: block;
  }

  ol,
  ul {
    list-style: none;
  }

  blockquote,
  q {
    quotes: none;
  }

  blockquote:before,
  blockquote:after,
  q:before,
  q:after {
    content: "";
    content: none;
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
`

const appResetMixin = css<StyledProps<IProps>>`
  html {
    /* stylelint-disable property-no-vendor-prefix */
    -ms-overflow-style: scrollbar; /* Set box-sizing so that the width is not affected by padding or border */
    -ms-text-size-adjust: 100%; /* Set default font family in all browsers. */
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); /* Makes 1rem = 10px */
    -webkit-text-size-adjust: 100%; /* Prevent adjustments of font size after orientation changes in IE on Windows Phone and in iOS. */
    /* stylelint-enable property-no-vendor-prefix */
    box-sizing: border-box;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }

  body {
    background-color: ${({theme}) => theme.palette.background.white};

    /* Hide overflow content */
    &.overflow {
      overflow: hidden;
    }

    /* Prevent focus outline on elements that can't be accessed via keyboard. */
    [tabindex="-1"]:focus {
      outline: none !important;
    }

    /* Hide overflow in IE */
    svg:not(:root) {
      overflow: hidden;
    }

    /* Hide elements has 'hidden' attribute in IE 10- */
    [hidden] {
      display: none !important;
    }

    ::-webkit-file-upload-button {
      /* stylelint-disable-next-line property-no-vendor-prefix */
      -webkit-appearance: button;
      font: inherit;
    }
  }
`

const typographyMixin = css<StyledProps<IProps>>`
  html {
    font-family: ${({theme}) =>
      theme.typography
        .fontFamily}; /* Prevent scrollbars to overlap content in IE11 and Edge */
    font-size: ${({theme}) =>
      toPercent(
        getValue(theme.typography.htmlFontSize),
        16
      )}; /* Set default tap highlight to be transparent in iOS. */
    font-style: normal;
    color: ${({theme}) => theme.palette.text.dark};
  }

  body {
    ${bodyMixin<IProps>()}
  }

  h1 {
    ${h1Mixin<IProps>()}
  }

  h2 {
    ${h2Mixin<IProps>()}
  }

  h3 {
    ${h3Mixin<IProps>()}
  }

  h4 {
    ${h4Mixin<IProps>()}
  }

  h5 {
    ${h5Mixin<IProps>()}
  }

  h6 {
    ${h6Mixin<IProps>()}
  }

  small {
    ${smallMixin<IProps>()}
  }

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.bold};

    h1,
    h2,
    h3,
    h4,
    h5,
    h6,
    caption {
      font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
    }
  }
`

const utilsMixin = css<StyledProps<IProps>>`
  .hidden {
    visibility: hidden;
  }

  .d-none {
    display: none;
  }

  .hidden-sm-down {
    ${({theme}) =>
      mediaBreakpointDown(
        "sm",
        theme.grid.breakpoints,
        `
      display: none;
    `
      )}
  }

  .text-center {
    text-align: center;
  }
`

const CssBaseline: GlobalStyleComponent<
  IProps,
  DefaultTheme
> = createGlobalStyle`
  ${genericResetMixin}

  ${appResetMixin}

  ${typographyMixin}

  ${utilsMixin}
`

export default CssBaseline
