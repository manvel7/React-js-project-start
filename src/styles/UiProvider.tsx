import React, {FunctionComponent} from "react"
import {ThemeProvider} from "styled-components"
import theme from "./theme"
import {CssBaseline} from "./index"

interface IProps {}

const UiProvider: FunctionComponent<IProps> = (props) => {
  const {children} = props

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  )
}

export default UiProvider
