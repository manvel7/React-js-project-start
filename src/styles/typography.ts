import {
  css,
  DefaultTheme,
  FlattenInterpolation,
  StyledProps,
  ThemedStyledProps
} from "styled-components"
import {pxToRem} from "../utils/unit"

const h1Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h1.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h1.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h1.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h1.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h2Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h2.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h2.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h2.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h2.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h3Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h3.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h3.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h3.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h3.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h4Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h4.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h4.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h4.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h4.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h5Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h5.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h5.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h5.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h5.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h6Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h6.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h6.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h6.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h6.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const bodyMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.body.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.body.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.lineHeight,
      theme.typography.htmlFontSize
    )};
`

const smallMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.caption.weight};
  letter-spacing: ${({theme}) =>
    theme.typography.variants.caption.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`
export {
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin,
  bodyMixin,
  smallMixin
}
