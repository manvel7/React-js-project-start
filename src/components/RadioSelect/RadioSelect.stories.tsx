import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import RadioItem from "./RadioItem"
import RadioSelect, {IProps} from "./RadioSelect"

const meta: Meta<IProps> = {
  title: "Components/RadioSelect",
  component: RadioSelect,
  argTypes: {
    onValueChange: {
      action: "onValueChange"
    }
  }
}

const Default: Story<IProps> = (args) => (
  <RadioSelect {...args}>
    <RadioItem label="First item" value="first" />
    <RadioItem label="Second item" value="second" />
    <RadioItem label="Third item" value="third" />
  </RadioSelect>
)

const WithoutLabels: Story<IProps> = (args) => (
  <RadioSelect {...args}>
    <RadioItem value="first" />
    <RadioItem value="second" />
    <RadioItem value="third" />
  </RadioSelect>
)

const WithAutoValues: Story<IProps> = (args) => (
  <RadioSelect {...args}>
    <RadioItem label="First item" />
    <RadioItem label="Second item" />
    <RadioItem label="Third item" />
  </RadioSelect>
)

const ErrorState = Default.bind({})
ErrorState.args = {
  error: true
}

const DisabledState = Default.bind({})
DisabledState.args = {
  disabled: true
}

export {Default, WithoutLabels, WithAutoValues, ErrorState, DisabledState}

export default meta
