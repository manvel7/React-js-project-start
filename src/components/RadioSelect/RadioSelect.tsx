import React, {
  ChangeEvent,
  Children,
  cloneElement,
  FunctionComponent,
  HTMLAttributes,
  useState
} from "react"
import RadioItem from "./RadioItem"
import RadioSelectWrapper from "./Wrapper"

export interface IProps extends HTMLAttributes<HTMLDivElement> {
  onValueChange?: (value: string) => void
  error?: boolean
  disabled?: boolean
}

const RadioSelect: FunctionComponent<IProps> = (props) => {
  const {children, disabled, onValueChange, error, ...rest} = props
  const [value, setValue] = useState<string>()

  const items = Children.map(children, (item: any, index) => {
    if (typeof item !== "object" || item.type !== RadioItem) {
      throw new TypeError(
        "RadioSelect children should only contain RadioItem elements"
      )
    }

    const {value: itemValue} = item.props

    return cloneElement(item, {
      onChange(event: ChangeEvent<HTMLInputElement>) {
        const {target} = event

        setValue(target.value)

        if (onValueChange) {
          onValueChange(target.value)
        }
      },
      checked: (itemValue || index.toString()) === value,
      value: itemValue || index.toString(),
      error,
      disabled
    })
  })

  return <RadioSelectWrapper {...rest}>{items}</RadioSelectWrapper>
}

export default RadioSelect
