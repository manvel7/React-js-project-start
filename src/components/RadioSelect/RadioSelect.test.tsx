import React from "react"
import {render} from "@testing-library/react"
import {UiProvider} from "../../styles"
import RadioItem from "./RadioItem"
import RadioSelect from "./RadioSelect"

describe("RadioItem", () => {
  describe("Default", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <RadioSelect>
            <RadioItem label="first" />
            <RadioItem label="second" />
            <RadioItem label="third" />
          </RadioSelect>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("ErrorState", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <RadioSelect error={true}>
            <RadioItem label="first" />
            <RadioItem label="second" />
            <RadioItem label="third" />
          </RadioSelect>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("DisabledState", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <RadioSelect disabled={true}>
            <RadioItem label="first" />
            <RadioItem label="second" />
            <RadioItem label="third" />
          </RadioSelect>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
})
