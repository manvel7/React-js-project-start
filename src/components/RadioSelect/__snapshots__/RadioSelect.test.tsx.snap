// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioItem Default should match snapshot 1`] = `
.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0.8rem;
}

.c1:last-child {
  margin-bottom: 0;
}

.c1 .radio-select__item {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FEFEFE;
  border: 0.2rem solid #001254;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 0.8rem;
  width: 1.6rem;
}

.c1 .radio-select__item:after {
  background-color: #001254;
  border-radius: 50%;
  content: "";
  display: block;
  height: 0.8rem;
  opacity: 0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: -webkit-transform 0.25s ease-in-out,opacity 0.2s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out,opacity 0.2s ease-in-out;
  transition: transform 0.25s ease-in-out,opacity 0.2s ease-in-out;
  width: 0.8rem;
}

.c0 {
  display: inline-block;
}

<div
  class="c0"
>
  <label
    class="c1"
    label="first"
  >
    <input
      class="c2"
      label="first"
      type="radio"
      value="0"
    />
    <div
      class="radio-select__item"
    />
    <span>
      first
    </span>
  </label>
  <label
    class="c1"
    label="second"
  >
    <input
      class="c2"
      label="second"
      type="radio"
      value="1"
    />
    <div
      class="radio-select__item"
    />
    <span>
      second
    </span>
  </label>
  <label
    class="c1"
    label="third"
  >
    <input
      class="c2"
      label="third"
      type="radio"
      value="2"
    />
    <div
      class="radio-select__item"
    />
    <span>
      third
    </span>
  </label>
</div>
`;

exports[`RadioItem DisabledState should match snapshot 1`] = `
.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0.8rem;
}

.c1:last-child {
  margin-bottom: 0;
}

.c1 .radio-select__item {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FEFEFE;
  border: 0.2rem solid #001254;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 0.8rem;
  width: 1.6rem;
}

.c1 .radio-select__item:after {
  background-color: #001254;
  border-radius: 50%;
  content: "";
  display: block;
  height: 0.8rem;
  opacity: 0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: -webkit-transform 0.25s ease-in-out,opacity 0.2s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out,opacity 0.2s ease-in-out;
  transition: transform 0.25s ease-in-out,opacity 0.2s ease-in-out;
  width: 0.8rem;
}

.c1 .radio-select__item {
  border-color: #DDDDDD;
}

.c1 .radio-select__item:after {
  background-color: #DDDDDD;
}

.c0 {
  display: inline-block;
}

<div
  class="c0"
>
  <label
    class="c1"
    disabled=""
    label="first"
  >
    <input
      class="c2"
      disabled=""
      label="first"
      type="radio"
      value="0"
    />
    <div
      class="radio-select__item"
    />
    <span>
      first
    </span>
  </label>
  <label
    class="c1"
    disabled=""
    label="second"
  >
    <input
      class="c2"
      disabled=""
      label="second"
      type="radio"
      value="1"
    />
    <div
      class="radio-select__item"
    />
    <span>
      second
    </span>
  </label>
  <label
    class="c1"
    disabled=""
    label="third"
  >
    <input
      class="c2"
      disabled=""
      label="third"
      type="radio"
      value="2"
    />
    <div
      class="radio-select__item"
    />
    <span>
      third
    </span>
  </label>
</div>
`;

exports[`RadioItem ErrorState should match snapshot 1`] = `
.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0.8rem;
}

.c1:last-child {
  margin-bottom: 0;
}

.c1 .radio-select__item {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FEFEFE;
  border: 0.2rem solid #001254;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 0.8rem;
  width: 1.6rem;
}

.c1 .radio-select__item:after {
  background-color: #001254;
  border-radius: 50%;
  content: "";
  display: block;
  height: 0.8rem;
  opacity: 0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: -webkit-transform 0.25s ease-in-out,opacity 0.2s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out,opacity 0.2s ease-in-out;
  transition: transform 0.25s ease-in-out,opacity 0.2s ease-in-out;
  width: 0.8rem;
}

.c1 .radio-select__item {
  border-color: #CA0000;
}

.c1 .radio-select__item:after {
  background-color: #CA0000;
}

.c0 {
  display: inline-block;
}

<div
  class="c0"
>
  <label
    class="c1"
    label="first"
  >
    <input
      class="c2"
      label="first"
      type="radio"
      value="0"
    />
    <div
      class="radio-select__item"
    />
    <span>
      first
    </span>
  </label>
  <label
    class="c1"
    label="second"
  >
    <input
      class="c2"
      label="second"
      type="radio"
      value="1"
    />
    <div
      class="radio-select__item"
    />
    <span>
      second
    </span>
  </label>
  <label
    class="c1"
    label="third"
  >
    <input
      class="c2"
      label="third"
      type="radio"
      value="2"
    />
    <div
      class="radio-select__item"
    />
    <span>
      third
    </span>
  </label>
</div>
`;
